### ISSUE TITLE
Culture invariant node with segment variant: "This item is not published"

### ISSUE THREAD

**Issue by PerplexDaniel** [umbraco/Umbraco-CMS#9074]
When a segment variant is created for a node that does not allow culture variants the Info tab shows "This item is not published", even though the node is published. This _only_ affects the backoffice UI, the node is still published and reachable.

![not published](https://user-images.githubusercontent.com/24297947/95062398-32943b80-06fd-11eb-8def-b65af44dc86f.png)
![segments not published](https://user-images.githubusercontent.com/24297947/95062437-3de76700-06fd-11eb-8469-575bdd86c0e5.png)

## Umbraco version

I am seeing this issue on Umbraco version: 8.7 + 8.8

### Steps to reproduce

1. Enable segmentation for a document type + 1 property type
2. Create a segment for a node of this document type using SetValue() on the IContent with some value for the "segment" parameter (e.g. `"tmp_segment"`)
3. Go to the Info tab and observe the message.

### Expected result

The URL for the Default variant and the segment variant should be the URL of the Default variant.
Segment variants do not have their own URLs like culture variants have.

### Actual result

Umbraco shows "This item is not published".

### Cause of bug
So this issue is [caused by the code](https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbcontentnodeinfo.directive.js#L324) for the `currentUrls` property in the `<umb-content-node-info>` directive. 
This code is checking the number of variants and if there is more than 1, assumes we are dealing with culture variation. However, now that segments are added this is no longer always the case.

```js
if (scope.node.variants.length > 1) {
    // nodes with variants
    scope.currentUrls = _.filter(scope.node.urls, (url) => (scope.currentVariant.language && scope.currentVariant.language.culture === url.culture));
} else {
    // invariant nodes
    scope.currentUrls = scope.node.urls;
}
```

So we end up in the `scope.node.variants.length > 1` case, which will then lead to an empty array of currentUrls because `currentVariant.language` is always null.

I will submit a PR shortly that will fix this issue by simplifying the `if/else` logic to 1 case that will handle everything.
Basically, we should just filter all URLs for the current language, but if there is not current language show all URLs.
This should work in all cases.
```js
scope.currentUrls = _.filter(scope.node.urls, (url) => (scope.currentVariant.language == null || scope.currentVariant.language.culture === url.culture));
```

**Comment by nul800sebastiaan** [comment]
Fixed with https://github.com/umbraco/Umbraco-CMS/pull/9075

Thanks very much @PerplexDaniel - great work! 🥇
